### PR TITLE
add CreateDirectChannel plugin api

### PR DESCRIPTION
**Pull request by cpanato**
#### Summary
add CreateDirectChannel plugin api

#### Checklist
- [x] Added or updated unit tests (required for all new features)

### CODE REVIEW

File: plugin/api.go [hunk 1]
```diff
@@ -42,6 +42,11 @@ type API interface {
 	// CreateUser creates a user.
 	CreateUser(user *model.User) (*model.User, *model.AppError)
 
+	// CreateDirectChannel creates a Direct channel.
```
Comment (hanzei) [inline plugin/api.go]:
```suggestion
	// CreateDirectChannel creates a direct message channel between two users.
```

Working taken from https://api.mattermost.com/#tag/channels%2Fpaths%2F~1channels~1direct%2Fpost

Comment (cpanato) [reply plugin/api.go]: done, thanks

File: plugin/api.go [hunk 2]
```diff
@@ -42,6 +42,11 @@ type API interface {
 	// CreateUser creates a user.
 	CreateUser(user *model.User) (*model.User, *model.AppError)
 
+	// CreateDirectChannel creates a direct message channel between two users.
```
Comment (jwilander) [inline plugin/api.go]:
```suggestion
	// CreateDirectChannel creates a direct message channel between two users.
	// If the channel was previously created, the existing channel is returned.
```

Comment (cpanato) [reply plugin/api.go]: done

**Comment by jwilander** [comment]
This one needs a rebase as well

**Pull request merged**